### PR TITLE
fix/LATTICE-2880-migration-fixes

### PR DESCRIPTION
**Pull request by njhlai**

### CODE REVIEW

File: conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt [hunk 1]
```diff
@@ -258,7 +260,30 @@ class ExternalDatabasePermissioner(
             columnAcls: List<Acl>,
             columnsById: Map<AclKey, TableColumn>
     ) {
-        updateTablePermissions(action, columnAcls, columnsById, TableType.VIEW)
+        // add acl/TableColumn associated to EdmConstants.ID_FQN (with uuid IdConstants.ID_ID.id)
+        val completedColumnAcls = columnAcls.toMutableList()
+        val completedColumnsById = columnsById.toMutableMap()
+        columnAcls.forEach { 
+            val idFqnAclKey = AclKey(it.aclKey[0], IdConstants.ID_ID.id)
+            val org_id = columnsById.getValue(it.aclKey).organizationId
```
Comment (UnsungHero97) [inline conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt:268]: Our style conventions are to use camelCase.

```suggestion
            val orgId = columnsById.getValue(it.aclKey).organizationId
```

File: conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt [hunk 2]
```diff
@@ -258,7 +260,30 @@ class ExternalDatabasePermissioner(
...
+            val readAces = it.aces.map { ace ->
+                Ace(ace.principal, EnumSet.of(Permission.READ))
+            }
+            
+            completedColumnsById.put(
+                idFqnAclKey, 
+                TableColumn(
+                    org_id,
```
Comment (UnsungHero97) [inline conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt:276]:
```suggestion
                    orgId, 
```

File: conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt [hunk 3]
```diff
@@ -258,7 +260,30 @@ class ExternalDatabasePermissioner(
             columnAcls: List<Acl>,
             columnsById: Map<AclKey, TableColumn>
     ) {
-        updateTablePermissions(action, columnAcls, columnsById, TableType.VIEW)
+        // add acl/TableColumn associated to EdmConstants.ID_FQN (with uuid IdConstants.ID_ID.id)
+        val completedColumnAcls = columnAcls.toMutableList()
+        val completedColumnsById = columnsById.toMutableMap()
+        columnAcls.forEach { 
+            val idFqnAclKey = AclKey(it.aclKey[0], IdConstants.ID_ID.id)
```
Comment (UnsungHero97) [inline conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt:267]: I think renaming this to something like `entityKeyIdAclKey` or `internalIdAclKey` would make it more clear that this is specifically for the entity key id.

Comment (njhlai) [reply conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt:267]: Sounds good, will get that in by tonight!

File: conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt [hunk 4]
```diff
@@ -369,13 +403,26 @@ class ExternalDatabasePermissioner(
 
             orgColumnAcls.forEach { columnAcl ->
                 val column = columnsById.getValue(columnAcl.aclKey)
-                val columnName = externalColumns.getValue(column.columnId)?.name ?: String()
-                val tableName = externalTables.getValue(column.tableId)?.name ?: String()
+                val columnName = if (securableObjectType == SecurableObjectType.OrganizationExternalDatabaseColumn) {
+                    externalColumns.getValue(column.tableId)?.name ?: String()
```
Comment (UnsungHero97) [inline conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt:407]: Shouldn't this be `columnId`?

```suggestion
                    externalColumns.getValue(column.columnId)?.name ?: String()
```

File: conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt [hunk 5]
```diff
@@ -369,13 +403,26 @@ class ExternalDatabasePermissioner(
...
+                } else {
+                    propertyTypes.getValue(column.tableId)?.type.toString() ?: String()
```
Comment (UnsungHero97) [inline conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt:409]: Shouldn't this be `columnId`?

```suggestion
                    propertyTypes.getValue(column.columnId)?.type.toString() ?: String()
```

File: conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt [hunk 6]
```diff
@@ -369,13 +403,26 @@ class ExternalDatabasePermissioner(
...
+                }
+                val tableName = if (securableObjectType == SecurableObjectType.OrganizationExternalDatabaseColumn) {
+                    externalTables.getValue(column.columnId)?.name ?: String()
```
Comment (UnsungHero97) [inline conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt:412]: Shouldn't this be `tableId`?

```suggestion
                    externalTables.getValue(column.tableId)?.name ?: String()
```

File: conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt [hunk 7]
```diff
@@ -369,13 +403,26 @@ class ExternalDatabasePermissioner(
...
+                } else {
+                    entitySets.getValue(column.columnId)?.name ?: String()
```
Comment (UnsungHero97) [inline conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt:414]: Shouldn't this be `tableId`?

```suggestion
                    entitySets.getValue(column.tableId)?.name ?: String()
```

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

**Review by UnsungHero97 — DISMISSED** [review]

**Review by UnsungHero97 — APPROVED** [review]

**Review by UnsungHero97 — APPROVED** [review]

**Review by and-carter — APPROVED** [review]

**Pull request merged**